src/movies.js

```diff
@@ -1,16 +1,140 @@
+    "duration": "3h 22min",
+    "genre": [
+      "Crime",
+      "Drama"
+    ],
+    "rate": 9
+  },
+  {
+    "title": "Schindler\"s List",
+    "year": 1993,
+    "director": "Steven Spielberg",
+    "duration": "3h 15min",
+    "genre": [
+      "Biography",
+      "Drama",
+      "History"
+    ],
+    "rate": 8.9
+  },
+  {
+    "title": "Saving Private Ryan",
+    "year": 1998,
+    "director": "Steven Spielberg",
+    "duration": "2h 49min",
+    "genre": [
+      "Drama",
+      "War"
+    ],
+    "rate": 8.6
+  }
+]
+
 // Iteration 1: All directors? - Get the array of all directors.
+
+const getAllDirectors = arr =>{
+  let movieDirectorsList= arr.map(name => name.director);
+  return movieDirectorsList;
+}
+getAllDirectors(movies);
+
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
-// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+// const getAllDirectors = (moviesList) => {
+//     const movieDirectorsList = moviesList.map((movie) => movie.director);
+//     const removedDuplicates = movieDirectorsList.reduce((acc, currentValue) => {
+//       if (acc.includes(currentValue)) {
+//         return acc;
+//       } else {
+//         return [...acc, currentValue];
+//       }
+//     }, []);
+//     return removedDuplicates;
+//   };
+
+//Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+
+//x is used even it does not make sense for now, it made me understand iteration better
+const howManyMovies = moviesList => {
+  let stevenDramaMovie= moviesList.filter((x) => x.director === "Steven Spielberg" && x.genre.includes("Drama"));
+  return stevenDramaMovie;
+}
+howManyMovies(movies);
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+const ratesAverage = arr =>{
+  if(arr.length ===0) {return 0};
+  let collectRate= arr.map(num=> num["rate"]);
+  let sumRate= collectRate.reduce((acc, curr) => acc + curr);
+  let averageRate= ((sumRate/arr.length).toFixed(2));
+  console.log(averageRate);
+}
+ratesAverage(movies);
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+const dramaMoviesRate = arr =>{
+  let dramaMovies= arr.map(film =>film["genre"].includes("Drama"));
+  if (dramaMovies.length === 0){return 0};
+  let collectDramaMovies= dramaMovies.map(film =>film["rate"]);
+  let average= collectDramaMovies.reduce((a,b)=> a+b);
+  let dramaAverageRate=((average/arr.length).toFixed(2));
+  console.log(dramaAverageRate);
+}
+ratesAverage(movies);
+  
+
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
 
+const orderByYear = arr =>{
+  let onlyYear= arr.map(date =>date.year);
+  let sortYear= onlyYear.sort((a,b)=> b-a);
+  return sortYear;
+}
+orderByYear(movies);
+
```

Comment:
Here's how you can write orderByYear, let me know if you have questions about it
```
function orderByYear(lotsOfMovies) {
  let newArray = [...lotsOfMovies];
  // spread operator is to make sure not to mutate the original array because .sort() does change/mutate the original array so always make sure you create a safe copy before sorting
  newArray.sort((a, b) => {
    if (a.year > b.year) {
      return 1;
    } else if (b.year > a.year) {
      return -1;
    } else {
      if (a.title > b.title) {
        return 1;
      } else if (b.title > a.title) {
        return -1;
      }
      return 0;
    }
  });
  return newArray;
}